aea/manager.py

```diff
@@ -551,20 +545,35 @@ def _make_config(
                 'Do not override any of {" ".join(self.AGENT_DO_NOT_OVERRIDE_VALUES)}'
             )
 
-        json_data = yaml_load_all(
-            AEABuilder.get_configuration_file_path(project_path).open()
+        agent_configuration_file_path: Path = AEABuilder.get_configuration_file_path(
+            project_path
         )
-        agent_config = self._update_dict(json_data[0], agent_overrides)
-
-        components_configs = {PublicId.from_json(obj): obj for obj in json_data[1:]}
-
+        loader = ConfigLoaders.from_package_type(PackageType.AGENT)
+        with agent_configuration_file_path.open() as fp:
+            agent_config: AgentConfig = loader.load(fp)
```

Comment:
we use `AgentConfig` instead of bare JSONs, the idea is that hopefully will make the code more robust, since we call the `agent_config.update` method to override the configurations, and there we can delegate the consistency checks